db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2038,7 +2040,10 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     WaitForPendingWrites();
 
     if (flush_reason != FlushReason::kErrorRecoveryRetryFlush &&
-        (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load())) {
+        (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) &&
+        (flush_options.min_size_to_flush > 0 &&
```

Comment:
I think it should be just `cfd->mem()->ApproximateMemoryUsageFast() >= flush_options.min_size_to_flush`.